#3 add option to run exomiser with hpo terms only

yaseminbridges opened the pull request:
Implemented an option to run Exomiser with ```--preset phenotype-only``` which only takes into account HPO terms when running analysis (no vcf required). This option should only be run with Exomiser version 13.2.0 onwards as previous versions have a much longer computational time for computing these results.

Methods dealing with new Exomiser 13.2.0 features, e.g. specification of results directory and file prefix, have also been implemented while retaining previous methods that are compatible with previous versions of Exomiser.

matentzn commented on src/pheval_exomiser/post_process/post_process_results_format.py:
```diff
@@ -191,9 +196,15 @@ def create_standardised_results(
+    default=False,
+    help="Specify if Exomiser was run with phenotype-only analysis.",
+)
 def post_process_exomiser_results(
```
What was the motivation to have a CLI for this? Shouldnt postprocessing be dealt with by pheval run? Or is this something internal for the exomiser team that is not relevant for PhEval?

yaseminbridges replied:
It's something internal for the Exomiser team that's not relevant for PhEval. When I do the benchmarking for Exomiser I use the HPC and as I haven't implemented an option to pheval run on the environment yet it's easier to use the separated commands for now

matentzn replied:
Cool thank you!

yaseminbridges commented on tests/exomiser_pheval_runner_test.py:
```diff
@@ -1,39 +1,42 @@
 """CLI Test """
-import logging
-import unittest
+# NOTE ALL TESTS IN THIS FILE FAIL WITH TOX
```
All the tests in the file fail when I run `tox`, but[ the invalid runner test](https://github.com/monarch-initiative/pheval.exomiser/blob/13fc03c895c19f79798acabada0383e95641c1e8/tests/exomiser_pheval_runner_test.py#L30) passes when I run it separately. I tried to figure out why the test was failing and it looks like a different error message is being returned regarding invalid requirements. 

The [exomiser cli ](https://github.com/monarch-initiative/pheval.exomiser/blob/13fc03c895c19f79798acabada0383e95641c1e8/tests/exomiser_pheval_runner_test.py#L18) test fails as it attempts to run all the pheval run steps and so will need some dummy data if this needs to be tested

matentzn replied:
Still that problem? Is it a good idea to comment out all tests?

julesjacobsen replied:
If these are actually testing Pheval functionality, they should move there and be deleted from this repo.

yaseminbridges commented:
Removed the manual path setting throughout the code for methods that are implemented in the runner. Refactored some variable names for consistency with the directory naming in PhEval. 

There shouldn't be any conflict now with the differences in the directory naming that @souzadevinicius mentioned in the meeting as I've removed the creation of these specific paths with the understanding that it is going to be handled by the MakeFile, i.e. the path that I am expecting to be passed to `pheval run` is `--output-dir ./exomiser-13.2.0/default-corpus1-default` .

Ultimately, this PR was to add in the option to run with the `phenotype-only` preset in Exomiser for a clearer comparison with Phen2Gene/other gene prioritisation softwares, so I have left in the reliance on the runner specific config etc for now but I know this is something that may be changed later

matentzn commented on pyproject.toml:
```diff
@@ -19,11 +19,19 @@ pyaml = "^21.10.1"
 oaklib = "^0.1.55"
 pyserde = "^0.9.7"
 docker = "^6.0.1"
+pheval = {git = "https://github.com/monarch-initiative/pheval.git"}
```
We should probably not do this.. But if it is very convenient, then make an issue to change this to a normal system. @souzadevinicius - how do you release pheval to pypi?

yaseminbridges replied:
As there isn't a recent release of PhEval on pypi this is the only way I can get things to pass with the qc

matentzn replied:
Can you request from Vinicius a workflow that makes a pypi release of pheval every time we merge to master? Does it make sense?

matentzn reviewed (approved):
src/pheval_exomiser/runner.py is what matters most, I didnt look to much at the rest, but LGTM

yaseminbridges commented:
> Still that problem? Is it a good idea to comment out all tests?

@matentzn I can't reply to this directly. These tests were used to test the CLI runner, I don't see how these could be implemented now that I have overwritten the methods within the Exomiser Runner. I'd need access to the exomiser data for it all to run without error which isn't possible. The other test checked if an error is raised when an invalid runner is passed but I don't think that's a test that should be done by a plugin but rather tested in PhEval instead

matentzn commented:
Sorry I thought we were in pheval repo! You are right ignore my comment.

yaseminbridges commented:
> Sorry I thought we were in pheval repo! You are right ignore my comment.

I will delete the file instead of hashing it all out to avoid confusion

julesjacobsen commented on src/pheval_exomiser/constants.py:
This is *only* used in post_process_results_format.py by trim_exomiser_result_filename() so I'd just leave it as a string in there.

julesjacobsen commented on tests/test_run.py:
```diff
@@ -90,6 +90,7 @@
         environment="local",
+        phenotype_only=False,
         path_to_exomiser_software_directory=Path(
             "/Users/yaseminbridges/exomiser/exomiser-cli-13.1.0"
```
probably don't want this like this. `/opt/exomiser/exomiser-cli-13.1.0` or whatever.

julesjacobsen commented on src/pheval_exomiser/prepare/create_batch_commands.py:
```diff
@@ -131,75 +157,140 @@ def write_command(
-        self, analysis_yaml, command_arguments: ExomiserCommandLineArguments
-    ) -> None:
-        """Write a docker command out for exomiser to run - including output option file specified."""
+    def write_phenotype_only_command(self, command_arguments: ExomiserCommandLineArguments):
```
All the write_thing and write_docker_thing methods are identical! The output is also identical (i.e. it's an Exomiser CLI string). Just do it once and pass in the relevant paths.
